test/test_cpp_extensions_aot.py

```diff
@@ -173,5 +173,213 @@ def test_rng(self):
         del copy2
         self.assertEqual(rng_extension.getInstanceCount(), 0)
 
+class TestCUDA_CSPRNG_Generator(common.TestCase):
+
+    def setUp(self):
+        import torch_test_cpp_extension.csprng as csprng_extension
+        super(TestCUDA_CSPRNG_Generator, self).setUp()
+        csprng_extension.registerOps()
+
+    @skipIfRocm
+    @unittest.skipIf(not TEST_CUDA, "CUDA not found")
+    def test_random(self):
+        import torch_test_cpp_extension.csprng as csprng_extension
+        gen = csprng_extension.create_CUDA_CSPRNG_Generator()
+        for dtype in [torch.bool, torch.uint8, torch.int8, torch.int16, 
+                      torch.int32, torch.int64, torch.float, torch.double]:
+            t = torch.empty(100, dtype=dtype, device='cuda').random_(generator=gen)
```

Comment:
You probably want *some* sort of test here. Randomized functions are a bit hard to test, but maybe you can find some inspiration in our existing tests for random numbers. A very mild test might be to test some statistical properties, like we do for some of our distributions tests.

Reply:
It's just very embarassing if, e.g., `random_` only ever zero-fills your tensor. Some basic sanity checks will help catch that..

Reply:
I see there are some statistical tests below. Does this test test anything that those tests below don't test?